lib/primitiv.gi

```diff
@@ -40,14 +43,35 @@ local s,fname,ind;
 end);
 
 BindGlobal("PRIMGrp",function(deg,nr)
-  PrimGrpLoad(deg);
+  local filename,strm,r,l;
   if nr>PRIMLENGTHS[deg] then
     Error("There are only ",PRIMLENGTHS[deg]," groups of degree ",deg,"\n");
   fi;
+  if deg > 4095 then
+    filename := Concatenation(GAPInfo.PackagesInfo.primgrp[1].InstallationPath,"/data/ExtendedPrimitiveGroupsData/PrimitiveGroups_", String(deg),"_", String(nr), ".g.gz");
+    strm:=InputTextFile(filename);;
+    if strm = fail then
```

Comment:
With the above changes, you can just test `filename`, and avoid using streams:
```suggestion
    if filename = fail then
```

Reply:
I had to put strm back so that I could read the contents of the file with ReadAll since it throws an error about a missing ; if I try to use Read. I changed the if statement to check if the filename has failed as you suggested, however.